### ISSUE TITLE
Reports of fatals when WooPayments and PayPal extensions are both enabled 

### ISSUE THREAD

**Issue by dmallory42** [Automattic/woocommerce-payments#9196]
### Describe the bug

There seems to be a fatal being thrown when both WooPayments and the PayPal plugin are enabled. Error trace here:

```
Fatal error: Uncaught Error: Call to undefined function WooCommerce\PayPalCommerce\Blocks\wc_terms_and_conditions_page_id() in /var/www/html/wp-content/plugins/woocommerce-paypal-payments/modules/ppcp-blocks/extensions.php on line 47 

Error: Call to undefined function WooCommerce\PayPalCommerce\Blocks\wc_terms_and_conditions_page_id() in /var/www/html/wp-content/plugins/woocommerce-paypal-payments/modules/ppcp-blocks/extensions.php on line 47 Call Stack: 0.0001 378048 1. {main}() /var/www/html/wp-admin/plugins.php:0 0.0001 384928 2. require_once('/var/www/html/wp-admin/admin.php') /var/www/html/wp-admin/plugins.php:10 0.0001 385632 3. require_once('/var/www/html/wp-load.php') /var/www/html/wp-admin/admin.php:34 0.0002 386296 4. require_once('/var/www/html/wp-config.php') /var/www/html/wp-load.php:50 0.0003 395496 5. require_once('/var/www/html/wp-settings.php') /var/www/html/wp-config.php:141 0.0738 13087912 6. do_action() /var/www/html/wp-settings.php:630 0.0738 13088288 7. 

WP_Hook->do_action() /var/www/html/wp-includes/plugin.php:517 0.0738 13088288 8. WP_Hook->apply_filters() /var/www/html/wp-includes/class-wp-hook.php:348 0.0740 13127496 9. WC_Payments::{closure:/var/www/html/wp-content/plugins/woocommerce-payments/includes/class-wc-payments.php:594-601}() /var/www/html/wp-includes/class-wp-hook.php:324 0.0779 13131696 

10. WC_Payments::maybe_init_woopay_direct_checkout() /var/www/html/wp-content/plugins/woocommerce-payments/includes/class-wc-payments.php:599 0.0779 13131696 
11. WC_Payments_Features::is_woopay_direct_checkout_enabled() /var/www/html/wp-content/plugins/woocommerce-payments/includes/class-wc-payments.php:1583 0.0779 13133656 12. WC_Payment_Gateways::instance() /var/www/html/wp-content/plugins/woocommerce-payments/includes/class-wc-payments-features.php:271 0.0779 13133736 

13. WC_Payment_Gateways->__construct() /var/www/html/wp-content/plugins/woocommerce/includes/class-wc-payment-gateways.php:49 0.0779 13133736 14. WC_Payment_Gateways->init() /var/www/html/wp-content/plugins/woocommerce/includes/class-wc-payment-gateways.php:76 0.0783 13211848 15. apply_filters() /var/www/html/wp-content/plugins/woocommerce/includes/class-wc-payment-gateways.php:94 0.0783 13212256 16. WP_Hook->apply_filters() /var/www/html/wp-includes/plugin.php:205 0.0783 13213384 17. WooCommerce\PayPalCommerce\WcGateway\WCGatewayModule::WooCommerce\PayPalCommerce\WcGateway\{closure:/var/www/html/wp-content/plugins/woocommerce-paypal-payments/modules/ppcp-wc-gateway/src/WCGatewayModule.php:510-572}() /var/www/html/wp-includes/class-wp-hook.php:324 0.0783 13213384 
14. 18. WooCommerce\PayPalCommerce\Vendor\Dhii\Container\CachingContainer->get() /var/www/html/wp-content/plugins/woocommerce-paypal-payments/modules/ppcp-wc-gateway/src/WCGatewayModule.php:511 0.0783 13214128 19. WooCommerce\PayPalCommerce\Vendor\Dhii\Container\CachingContainer->getCached() /var/www/html/wp-content/plugins/woocommerce-paypal-payments/lib/packages/Dhii/Container/CachingContainer.php:62 0.0783 13214128 20. WooCommerce\PayPalCommerce\Vendor\Dhii\Container\CachingContainer->invokeGenerator() /var/www/html/wp-content/plugins/woocommerce-paypal-payments/lib/packages/Dhii/Container/CachingContainer.php:128 0.0783 13214128 21. WooCommerce\PayPalCommerce\Vendor\Dhii\Container\CachingContainer->WooCommerce\PayPalCommerce\Vendor\Dhii\Container\{closure:/var/www/html/wp-content/plugins/woocommerce-paypal-payments/lib/packages/Dhii/Container/CachingContainer.php:60-62}() /var/www/html/wp-content/plugins/woocommerce-paypal-payments/lib/packages/Dhii/Container/CachingContainer.php:161 0.0783 13214128 22. WooCommerce\PayPalCommerce\Vendor\Dhii\Container\CompositeContainer->get() /var/www/html/wp-content/plugins/woocommerce-paypal-payments/lib/packages/Dhii/Container/CachingContainer.php:61 0.0783 13214128 23. WooCommerce\PayPalCommerce\Vendor\Dhii\Container\DelegatingContainer->get() /var/www/html/wp-content/plugins/woocommerce-paypal-payments/lib/packages/Dhii/Container/CompositeContainer.php:53 0.0784 13214128 24. WooCommerce\PayPalCommerce\Vendor\Dhii\Container\DelegatingContainer->invokeFactory() /var/www/html/wp-content/plugins/woocommerce-paypal-payments/lib/packages/Dhii/Container/DelegatingContainer.php:56 0.0784 13214128 25. WooCommerce\PayPalCommerce\WcGateway\WCGatewayModule::WooCommerce\PayPalCommerce\WcGateway\{closure:/var/www/html/wp-content/plugins/woocommerce-paypal-payments/modules/ppcp-wc-gateway/services.php:75-112}() /var/www/html/wp-content/plugins/woocommerce-paypal-payments/lib/packages/Dhii/Container/DelegatingContainer.php:117 0.0798 13294680 26. WooCommerce\PayPalCommerce\Vendor\Dhii\Container\DelegatingContainer->get() /var/www/html/wp-content/plugins/woocommerce-paypal-payments/modules/ppcp-wc-gateway/services.php:77 0.0798 13294680 27. WooCommerce\PayPalCommerce\Vendor\Dhii\Container\DelegatingContainer->invokeFactory() /var/www/html/wp-content/plugins/woocommerce-paypal-payments/lib/packages/Dhii/Container/DelegatingContainer.php:56 0.0798 13294680 28. WooCommerce\PayPalCommerce\WcGateway\WCGatewayModule::WooCommerce\PayPalCommerce\WcGateway\{closure:/var/www/html/wp-content/plugins/woocommerce-paypal-payments/modules/ppcp-wc-gateway/services.php:315-336}() /var/www/html/wp-content/plugins/woocommerce-paypal-payments/lib/packages/Dhii/Container/DelegatingContainer.php:117 0.0798 13294736 29. WooCommerce\PayPalCommerce\Vendor\Dhii\Container\DelegatingContainer->get() /var/www/html/wp-content/plugins/woocommerce-paypal-payments/modules/ppcp-wc-gateway/services.php:318 0.0798 13294736 30. WooCommerce\PayPalCommerce\Vendor\Dhii\Container\DelegatingContainer->invokeExtension() /var/www/html/wp-content/plugins/woocommerce-paypal-payments/lib/packages/Dhii/Container/DelegatingContainer.php:74 0.0798 13294736 31. WooCommerce\PayPalCommerce\Vendor\Dhii\Container\CompositeCachingServiceProvider->WooCommerce\PayPalCommerce\Vendor\Dhii\Container\{closure:/var/www/html/wp-content/plugins/woocommerce-paypal-payments/lib/packages/Dhii/Container/CompositeCachingServiceProvider.php:130-135}() /var/www/html/wp-content/plugins/woocommerce-paypal-payments/lib/packages/Dhii/Container/DelegatingContainer.php:144 0.0798 13294736 32. WooCommerce\PayPalCommerce\Vendor\Dhii\Container\CompositeCachingServiceProvider->WooCommerce\PayPalCommerce\Vendor\Dhii\Container\{closure:/var/www/html/wp-content/plugins/woocommerce-paypal-payments/lib/packages/Dhii/Container/CompositeCachingServiceProvider.php:130-135}() /var/www/html/wp-content/plugins/woocommerce-paypal-payments/lib/packages/Dhii/Container/CompositeCachingServiceProvider.php:131 0.0798 13294736 33. WooCommerce\PayPalCommerce\Vendor\Dhii\Container\CompositeCachingServiceProvider->WooCommerce\PayPalCommerce\Vendor\Dhii\Container\{closure:/var/www/html/wp-content/plugins/woocommerce-paypal-payments/lib/packages/Dhii/Container/CompositeCachingServiceProvider.php:130-135}() /var/www/html/wp-content/plugins/woocommerce-paypal-payments/lib/packages/Dhii/Container/CompositeCachingServiceProvider.php:131 0.0798 13294736 34. WooCommerce\PayPalCommerce\Vendor\Dhii\Container\CompositeCachingServiceProvider->WooCommerce\PayPalCommerce\Vendor\Dhii\Container\{closure:/var/www/html/wp-content/plugins/woocommerce-paypal-payments/lib/packages/Dhii/Container/CompositeCachingServiceProvider.php:130-135}() /var/www/html/wp-content/plugins/woocommerce-paypal-payments/lib/packages/Dhii/Container/CompositeCachingServiceProvider.php:131 0.0801 13300960 35. WooCommerce\PayPalCommerce\Blocks\BlocksModule->WooCommerce\PayPalCommerce\Blocks\{closure:/var/www/html/wp-content/plugins/woocommerce-paypal-payments/modules/ppcp-blocks/extensions.php:31-73}() /var/www/html/wp-content/plugins/woocommerce-paypal-payments/lib/packages/Dhii/Container/CompositeCachingServiceProvider.php:132
There has been a critical error on this website. Please check your site admin email inbox for instructions.

[Learn more about troubleshooting WordPress.](https://wordpress.org/documentation/article/faq-troubleshooting/)
```

As far as I can see, this issue is coming from the `is_woopay_direct_checkout_enabled` function, and was introduced in the 8.0.0 release: https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments-features.php#L268

It's possibly (but not confirmed yet) introduced here: https://github.com/Automattic/woocommerce-payments/pull/9159

From what I can gather, I think the issue is this:

- The `is_woopay_direct_checkout_enabled` initialises an instance of `WC_Payment_Gateways` - which would include the Paypal gateway if Paypal extension is installed
- Upon initialisation, PayPal gateway calls the function `wc_terms_and_conditions_page_id()` - a WooCommerce Core function
- This causes the fatal to be thrown (unsure why yet but I suspect it's either a namespacing issue, or that the function is being called before being defined in the execution lifecycle).

### To Reproduce

1. Install both the WooPayments and Woocommerce PayPal extensions. (You possibly also need to onboard a WooPayments account).
2. The fatal will get thrown. 

### Actual behavior

A fatal error is thrown on the site, and can only be fixed by disabling either extension or rolling back WooPayments.

**Comment by ClassicRKR27** [comment]
I've got another report here: 8555609-zd-a8c

**Comment by asumaran** [comment]
We are [reverting the changes](https://github.com/Automattic/woocommerce-payments/issues/9196) that introduced the fatal error. A patch release is incoming.

**Comment by oaratovskyi** [comment]
Closed by #9200

**Comment by oaratovskyi** [comment]
oh, and it was released in 8.0.1

**Comment by asumaran** [comment]
Just wanted to add in case you don't know, you can find the WooPayments 8.0.1 release [here](https://github.com/Automattic/woocommerce-payments/releases/tag/8.0.1) or even better here https://wordpress.org/plugins/woocommerce-payments/
